code/procs/gamehelpers.dm

```diff
@@ -452,6 +452,28 @@ var/obj/item/dummy/click_dummy = new
 				. += R
 				break
 
+/proc/get_areas_with_unblocked_turfs(var/areatype)
```

Comment:
I don't like how slow this is gonna be especially when run like 7 times, it only happens once so it's not a big deal but it makes me nervous. (A marginally faster way would be getting a list of wanted areas and a list of unwanted areas and then cheeck that inside a proc similar to this one.)